File: models/UserProfile/user_profile.py
from models import db

class UserProfile(db.Model):
    __tablename__ = 'user_profiles'
    
    id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
    first_name = db.Column(db.String(50), nullable=False, default="")
    last_name = db.Column(db.String(50), nullable=False, default="")
    bio = db.Column(db.Text, default="")
    profile_picture = db.Column(db.String(255), default="")
    
    user = db.relationship('User', back_populates='profile')

File: models/user_profile.go
package models

type UserProfile struct {
	ID             uint   `gorm:"primaryKey" json:"id"`
	UserID         uint   `gorm:"not null" json:"user_id"`
	FirstName      string `gorm:"size:80;not null" json:"first_name"`
	LastName       string `gorm:"size:80;not null" json:"last_name"`
	Bio            string `gorm:"type:text" json:"bio"`
	ProfilePicture string `gorm:"size:255" json:"profile_picture"`

	User *User `gorm:"foreignKey:UserID" json:"user,omitempty"`
}
